src/bitmat.c

```diff
@@ -1,586 +1,598 @@
+              if ((p[1] & ~NOREPORT) == (*(mat->vecs[k]-1) & ~NOREPORT)){
+                *(mat->vecs[k]-1) |= NOREPORT;}
+            }
+          }                     /* mark subsets if they have */
+        }                       /* the same support as the superset */
+        red->vecs[red->cnt++] = p+2;
+        *p = *(mat->vecs[k]-2); /* store and count the intersection */
+        p += ((mat->len < 0) ? p[1] : mat->len) +2;
+      }                         /* advance the vector pointer */
+      if (red->cnt <= 0) continue; /* if the matrix is empty, cont. */
+      ao->rows[depth-1] = *(mat->vecs[i]-2);
+      cnt += k = _search(ao, red, depth,mode);
+      if (k < 0) break;         /* recursively search for a submatrix */
+    }                           /* (i.e., frequent itemsets) */
+    free(vecs); free(red);      /* delete the work buffers */
+    #ifdef BENCH                /* if benchmark version */
+    ao->mcur -= sizeof(REDMAT) +(mat->cnt-2) *sizeof(int*)
+              + m *sizeof(int); /* compute current memory usage */
+    #endif
+    if (i >= n) return -1;      /* check for a search error */
+  }  /* if ((depth < ao->max) .. */
+
+  /* --- report (and record) found item sets --- */
+  if (depth >= ao->min) {       /* if minimum number of rows reached */
+    for (i = 0; i < mat->cnt; i++) {
+      p = mat->vecs[i]-2;       /* traverse the bit vectors */
+      ao->rows[depth-1] = p[0]; /* note the last row identifier */
+      if      (!ao->res)        n = 0; /* if free item sets, report */
+      else if (p[1] & NOREPORT) n = 1; /* skip non-closed/non-maximal */
+      else n = _exists(ao->res, ao->rows, depth, p[1] & ~NOREPORT);
+      
+      if (n < 0) return -1;     /* record closed/generator/maximal item set */
+      if (n > 0) continue;      /* if the item set qualifies */
+      k = p[1] & ~NOREPORT;     /* get the item set support */
+      ao->report(ao->rows, depth, (mat->len < 0) ? k:-k, p+2, ao->data);
+      cnt++;                    /* report the solutions found */
+    }                           /* and count the item sets */
+  }
+  return cnt;                   /* return number of item sets */
+}  /* _search() */
+
+/*--------------------------------------------------------------------*/
+
+static int _buffers (BITMAT *bm, int mode)
+{                               /* --- add buffers to created matrix */
+  bm->buf = (int*)malloc((BLKSIZE+1) *sizeof(int)) +1;
+  if (!bm->buf)   { bm_delete(bm); return -1; }
+  if ((mode != BM_CLOSED)&&(mode != BM_GENERATOR)) return 0;
+  bm->supps = (int*)malloc((BLKSIZE << BM_SHIFT) *sizeof(int));
+  if (!bm->supps) { bm_delete(bm); return -1; }
+  return 0;                     /* allocate additional buffers */
+}  /* _buffers() */
+
+/*--------------------------------------------------------------------*/
+
+int bm_allone (BITMAT *bm, int mode, int supp, int min, int max,
+               BMREPFN report, void *data,int tacnt)

```

Comment:
/* Makh2018: Adding number of transactions "tacnt" as parameter to extract full support items*/